### PR TITLE
Fix S3.GetBucketLocation by editing response body

### PR DESCRIPTION
**Pull request by adam-fowler**
Fixes https://github.com/swift-aws/aws-sdk-swift/issues/157
This pull request needs https://github.com/swift-aws/aws-sdk-swift-core/pull/106

Also
- renamed AWSRequestMiddleware to AWSServiceMiddleware
- S3 GET requests don't mangle the hostname if it isn't amazonaws.com as is done for all other requests

### CODE REVIEW

File: Sources/AWSSDKSwift/Middlewares/S3/S3RequestMiddleware.swift [hunk 1]
```diff
@@ -15,6 +15,7 @@ public struct S3RequestMiddleware: AWSRequestMiddleware {
 
         switch request.httpMethod.lowercased() {
         case "get":
+            guard let host = request.url.host, host.contains("amazonaws.com") else { break }
```
Comment (jonnymacs) [inline Sources/AWSSDKSwift/Middlewares/S3/S3RequestMiddleware.swift]: 🤔 on line 21 we check for this, and support when it's not true. Is there a valid use case where the host does not contain `amazonaws`?

Comment (jonnymacs) [reply Sources/AWSSDKSwift/Middlewares/S3/S3RequestMiddleware.swift]: right. So then what is the purpose of the code on lines 21-26 after this change?

Comment (adam-fowler) [reply Sources/AWSSDKSwift/Middlewares/S3/S3RequestMiddleware.swift]: I guess they can be removed, I'll do that

Comment (adam-fowler) [reply Sources/AWSSDKSwift/Middlewares/S3/S3RequestMiddleware.swift]: Removed this change so we can submit this PR. Will add another PR for this

**Pull request merged**